### PR TITLE
bug(firestore): match fb firestore behaviour

### PR DESCRIPTION
**Pull request by russellwheatley**
A couple of changes to the RNFB Firestore package that make it behave the same as its Firestore web counterpart.

- Removed the condition that stopped validation if there is more than one `orderBy` parameter.
- Cannot have a `where` query `fieldPath` parameter that matches any `orderBy` parameter.

### Checklist

- [ ] Supports `Android`
- [ ] Supports `iOS`
- [ ] `e2e` tests added or updated in packages/**/e2e
- [ ] Flow types updated
- [ ] Typescript types updated

### Release Plan

[CATEGORY][type] [LOCATION] - Message

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Salakar** [comment]
Is there still work to be done on this one? Some failures on CI;

image

**Comment by russellwheatley** [comment]
Yep, there is. I got my test suite working locally again last night so I should be able to request review at some point today and get my 2 PRs reviewed 👍

File: packages/firestore/e2e/Query/query.e2e.js [hunk 1]
```diff
@@ -27,4 +27,46 @@ describe('FirestoreQuery/FirestoreQueryModifiers', () => {
     queryAfter._modifiers._orders.length.should.equal(1);
     queryAfter._modifiers._filters.length.should.equal(1);
   });
+
+  it('Throws if where equality operator is invoked, and the where fieldPath parameter matches any orderBy parameter ', async () => {
```
Comment (Salakar) [inline packages/firestore/e2e/Query/query.e2e.js]:
```suggestion
  it('throws if where equality operator is invoked, and the where fieldPath parameter matches any orderBy parameter', async () => {
```

File: packages/firestore/e2e/Query/query.e2e.js [hunk 2]
```diff
@@ -27,4 +27,46 @@ describe('FirestoreQuery/FirestoreQueryModifiers', () => {
...
+        .orderBy('bar')
+        .orderBy('foo')
+        .limit(1)
+        .endAt(2);
+    } catch (error) {
+      error.message.should.containEql('Invalid query');
+    }
+  });
+
+  it('Throws if where inequality operator is invoked, and the where fieldPath does not match initial orderBy parameter', async () => {
```
Comment (Salakar) [inline packages/firestore/e2e/Query/query.e2e.js]:
```suggestion
  it('throws if where inequality operator is invoked, and the where fieldPath does not match initial orderBy parameter', async () => {
```

**Pull request merged**